nixos/modules/services/hardware/dockd.nix

```diff
@@ -29,65 +31,131 @@ in
       description = ''
         Script to be executed when the ThinkPad is removed from the dock.
-        If defined, the "undockHookPath" option will be ignored.
+        If is set to null (the default) then create dummy script to
+        fulfill the expectations of dockd.
+
+        Example:  "echo undocked"
+
+        This option is mutually exclusive to the "undockHookPath" option.
       '';
     };
 
   };
   ##################################################
   config =
-  let
-    # override the paths on which the dockd executable
-    # should lookup for the configuration and hook scripts
-    dockd =
-     pkgs.dockd.override {
-       inherit (cfg) dockedConfPath undockedConfPath;
-       dockHookPath = (if cfg.dockHook != ""
-                       then pkgs.writeScript "dock.hook"
-                       else cfg.dockHookPath);
-       undockHookPath = (if cfg.undockHook != ""
-                         then pkgs.writeScript "undock.hook" cfg.undockHook
-                         else cfg.undockHookPath);
-     };
-  in mkIf cfg.enable {
+    let
+       # override the paths on which the dockd executable
+       # should lookup for the configuration and hook scripts
+       dockd =
+        # configure a default script content if the dockHook or undockHook hook is
+        # set to null, specified that the user is not interested on doing nothing
+        # with the hooks
+        let
+          dockHook = if cfg.dockHook == null then "exit 0" else cfg.dockHook;
+          undockHook = if cfg.undockHook == null then "exit 0" else cfg.undockHook;
+          inherit(pkgs) writeScript;
+        in
+        pkgs.dockd.override {
+          inherit (cfg) dockedConfPath undockedConfPath;
+          dockHookPath = (if dockHook  != ""
+                          then writeScript "dock.hook" dockHook
+                          else cfg.dockHookPath);
+          undockHookPath = (if undockHook  != ""
+                            then writeScript "undock.hook" undockHook
+                            else cfg.undockHookPath);
+        };
+    in
+    mkIf cfg.enable {
+    assertions = [
+      # verify that only one of (dockHook | dockHookPath) is defined
+      {
+        assertion =
+         if ((cfg.dockHook != null && cfg.dockHook != "") && (cfg.dockHookPath != ""))
+         then false else true;
+        message = ''
+        You have to define ONLY one of "dockHook" or "dockHookPath"
+        '';
+      }
+      # verify that only one of (undockHook | undockHookPath) is defined
+      {
+        assertion =
+          if ((cfg.undockHook != null && cfg.undockHook != "") && (cfg.undockHookPath != ""))
+          then false else true;
+        message = ''
+        You have to define ONLY one of "undockHook" or "undockHookPath"
+        '';
+      }
+      # if the dockHookPath is used, verify the path exists
+      {
+        assertion =
+          if cfg.dockHookPath != "" then
+          (builtins.pathExists cfg.dockHookPath) else true;
```

Comment:
`builtins.pathExists` will check the path on the machine that evaluates the configuration, which may not be the machine that runs the configuration.
This would break for example a setup where a company has CI for its fleet of laptops. Also, it doesn't work in `restrict-eval` enabled.

Reply:
:thinking:  yeah, I never considered that when I added this assertion.